lib/tasks/rummager.rake

```diff
@@ -45,6 +45,25 @@ namespace :rummager do
       end
       puts "Complete."
     end
+
+    desc "indexes all content belonging to a model"
```

Comment:
I think we should probably remove the detailed guides one actually.

This is what it does:
```ruby
      index = Whitehall::SearchIndex.for(:detailed_guides)
      index.add_batch(Whitehall.detailed_guidance_search_index)
      index.commit
```

`Whitehall.detailed_guidance_search_index` is just a global method that wraps `DetailedGuide.search_index`. Once upon a time this method was called `specialist_search_index` and it had an implementation that did something (https://github.com/alphagov/whitehall/commit/0ad448038ddcf96c9b2bc934a84d26694244bfa1). Since then, both the implementation and the name have changed, and what used to be a helper for a particular index is now a useless format specific method that serves no purpose.

Keep in mind that `search_index` is not a search index, it's actually just a collection of documents. In 2013 it meant "the contents of an index" but that meaning has been lost. I'll look at renaming this in a follow up PR as I'm not sure how much of whitehall uses this confusing name, and this is what usually happens when I try to rename something in whitehall:
![image](http://66.media.tumblr.com/04807dd75132480d0c26cbcc09f85d3e/tumblr_inline_o4vdewfAEa1raprkq_500.gif)

Calling `add_batch` directly means we make one request to the underlying `store`, instead of making many 1-document batches. The store is defined in the deprecated `rummageable` gem. See: https://github.com/alphagov/rummageable/blob/master/lib/rummageable.rb#L19

This ultimately means we make less requests to rummager, but I believe rummager still indexes them one by one in elasticsearch.

For this PR I think we might as well switch all the implementations to `add_batch`, but I'd like to get rid of the detailed-guide-specific stuff in favour of something that works for all formats.